taskdart.dart

```diff
@@ -0,0 +1,148 @@
+/*Create a new Dart file (e.g., task_list.dart) to build the task list app.
+Implement a menu-driven interface that allows users to, view update, and delete tasks using the Task class.*/
+import 'dart:io';
+
+void main() {
+  //Task task1 = Task(1, 'Buy groceries', 'Buy milk, eggs, and bread', false);
+  //Task task2 = Task(2, 'Read a book', 'Read "The Catcher in the Rye"', true);
+
+  //print('Task 1: ${task1.title}, Completed: ${task1.isCompleted}');
+  //print('Task 2: ${task2.title}, Completed: ${task2.isCompleted}');
+  app taskListApp = app();
+  taskListApp.ap();
+}
+
+class Task {
+  // task class
+  int id;
+  String title;
+  String description;
+  bool isCompleted;
+
+  Task(
+      {required this.id,
+      required this.title,
+      required this.description,
+      required this.isCompleted});
+}
+
+class app {
+  List<Task> tasks = [];
+  // we want to put a menu for the user to choose or view, update, deleteing tasks
+  // we want to make a method for the user the user to choose to add and so on
+
+  void menu() {
+    print('Task list app');
+    print('please choose what you want to do');
+    print('1. add tasks');
+    print('2. view tasks');
+    print('3. update tasks');
+    print('4. delete tasks');
+    print('5. exit ');
+  }
+
+  // now we want to make methods for what the user want to choose
+  // we want something to read from tje user so we will user dart io:
+
+  void add() {
+    print('Please enther the task deatils');
+    stdout.write('please enter the task title: ');
+    String title = stdin.readLineSync()!;
+    stdout.write('please enter the task description: ');
+    String description = stdin.readLineSync()!;
```

Comment:
Both inputs need null checking
```suggestion
    stdout.write('please enter the task title: ');
    String title = stdin.readLineSync()!;
    if(title.isEmpty)
    {
        print("Cannot have empty title");
        return 404;
    }
    stdout.write('please enter the task description: ');
    String description = stdin.readLineSync()!;
    if(description.isEmpty)
    {
        print("Cannot have empty description");
        return 404;
    }
```